src/util/XDRCereal.h

```diff
@@ -17,6 +17,16 @@
 
 using namespace std::placeholders;
 
+template <uint32_t N>
+void
+cereal_override(cereal::JSONOutputArchive& ar, const xdr::xstring<N>& s,
+                const char* field)
+{
+    std::stringstream out;
```

Comment:
I don't think you need the extra buffer as all you want is to use the `std::string` flavor.

I think you just need this:

```
template <uint32_t N>
void
cereal_override(cereal::JSONOutputArchive& ar, const xdr::xstring<N>& s,
                const char* field)
{
    xdr::archive(ar, static_cast<std::string const&>(s), field);
}
```

Reply:
That's much simpler! Fixed.